### DISCUSSION TITLE
Are the environment variables read in at runtime?

### DISCUSSION THREAD

**Discussion by jeremeevans** [nuxt/nuxt#18202] [Questions]
If I create a .env file and build a docker image and run the container locally, the values I enter in nuxt.config.js work. So, entered here:

```
...
        privateRuntimeConfig: {
            APIURL: env.UXAPIURL,
            APIMKEY: env.UXAPIMKEY,
        },
...
```

Used here:

```
import config from "#config"
...
    console.log("Our API Url is ", config.APIURL)
...
```

When we go and build it in CICD and deploy it to Kubernetes, we want to re-use the same docker image across environments, so we'd really like to read in those settings at runtime. We build the docker image without the .env file present and added the environment variables to the node process. Sadly, when running on the K8s, config.APIURL is coming across as undefined.

We did inspect the process and saw that the environment variables were specified correctly there. Should we maybe not be using the .env file and `#config` and just use the node process.env?

**Comment by jeremeevans** [comment]
Answer seems to be no, they are not. So... if you're building an image to use in several environments, you need to gather things from process.env. This has me thinking the built in config mechanism is of little value in an enterprise environment.

**Comment by danielroe** [comment]
Currently you need to match the environment variable to the key of runtime config. For example:

```js
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  publicRuntimeConfig: {
    BASE_URL: process.env.BASE_URL || 'https://myapi.com'
  }
})
```

Then:
```
yarn nuxi build
BASE_URL=https://otherapi.com node .output/server/index.mjs
```
